music21/roman.py

```diff
@@ -3089,6 +3091,28 @@ def _updatePitches(self):
                 f'_updatePitches() was unable to derive pitches from the figure: {self.figure!r}'
             )  # pragma: no cover
 
+    def transpose(self, value, *, inPlace=False) -> t.Optional[T]:
+        '''
+        Overrides :meth:`~music21.harmony.Harmony.transpose` so that `key`
+        attribute is transposed as well.
+
+        >>> rn = roman.RomanNumeral('I', 'C')
+        >>> rn
+        <music21.roman.RomanNumeral I in C major>
+        >>> rn.transpose(4)
+        <music21.roman.RomanNumeral I in E major>
+        >>> rn.transpose(-4, inPlace=True)
+        >>> rn
+        <music21.roman.RomanNumeral I in A- major>
+        '''
+        post = super().transpose(value, inPlace=inPlace)
+        if not inPlace:
+            post.key = self.key.transpose(value, inPlace=False)
+            return post
+        else:
+            self.key = self.key.transpose(value, inPlace=False)
```

Comment:
why couldn't this be `self.key.transpose(value, inPlace=True)` ?

Reply:
That's what I had in my initial commit (a40e41a7f04b829ef02e9d8cb66a7dea1a78af70) but I changed it based on the discussion in #1413 and your feedback at https://github.com/cuthbertLab/music21/issues/1413#issuecomment-1247205169.

Reply:
ah okay -- yes, we solved the problem in RomanText but not elsewhere, my bad.